app/packages/core/src/components/Modal/Group/DynamicGroup/carousel/DynamicGroupsFlashlightWrapper.tsx

```diff
@@ -0,0 +1,302 @@
+            : node.aspectRatio,
+          id: (node.sample as any)._id as string,
+        },
+      ];
+    }
+
+    return acc;
+  }, []);
+
+export const DynamicGroupsFlashlightWrapper = () => {
+  const id = useId();
+  const pageCount = useRef(0);
+
+  const { groupByFieldValue } = useGroupContext();
+
+  const { data, hasNext, loadNext } = usePaginationFragment(
+    foq.paginateGroupPaginationFragment,
+    useRecoilValue(
+      fos.dynamicGroupPaginationFragment({
+        fieldOrExpression: groupByFieldValue!,
+      })
+    )
+  );
+
+  // todo: support pcd
+  const samples = useMemo(
+    () => ({
+      ...data.samples,
+      edges: data.samples.edges.filter(
+        (s) => s.node.sample._media_type !== "point-cloud"
+      ),
+    }),
+    [data]
+  );
+
+  const store = fos.useLookerStore();
+  const opts = fos.useLookerOptions(true);
+  const createLooker = fos.useCreateLooker(
+    true,
+    true,
+    {
+      ...opts,
+    },
+    true
+  );
+
+  const hasNextRef = useRef(true);
+  hasNextRef.current = hasNext;
+  const countRef = useRef(0);
+  const handleError = useErrorHandler();
+
+  const resolveRef = useRef<((value: Response<number>) => void) | null>(null);
+  const nextRef = useRef(loadNext);
+  nextRef.current = loadNext;
+
+  useEffect(() => {
+    if (resolveRef.current && countRef.current !== samples.edges.length) {
+      pageCount.current += 1;
+      resolveRef.current({
+        items: process(
+          countRef,
+          store,
+          false,
+          samples.edges.slice(countRef.current)
+        ),
+        nextRequestKey: pageCount.current,
+      });
+      countRef.current = samples.edges.length;
+      resolveRef.current = null;
+    }
+  }, [samples]);
+
+  const setSample = fos.useSetExpandedSample();
+
+  const select = fos.useSelectSample();
+  const selectSample = useRef(select);
+  const flashlightRef = useRef<Flashlight<number>>();
+  selectSample.current = select;
+
+  const navigationCallback = useRecoilCallback(
+    ({ snapshot }) =>
+      async (isPrevious) => {
+        const flashlight = flashlightRef.current;
+
+        if (!flashlight) {
+          return;
+        }
+
+        const modalSample = await snapshot.getPromise(fos.modal);
+
+        if (!modalSample) {
+          return;
+        }
+
+        const currentSampleIndex =
+          flashlight.itemIndexes[modalSample?.sample._id];
+        const nextSampleIndex = currentSampleIndex + (isPrevious ? -1 : 1);
+        const nextSampleId = store.indices.get(nextSampleIndex);
+
+        if (!nextSampleId) {
+          return;
+        }
+
+        const nextSample = store.samples.get(nextSampleId);
+        nextSample && setSample(nextSample);
+
+        // todo: very unstable, just ideating auto scroll based on quickstart groups params
+        if (flashlightRef.current) {
+          const newLeft = isPrevious
+            ? flashlightRef.current?.element.scrollLeft - 325 * 3
```

Comment:
Are these relative to the image sizes? If so I'm guessing we'll see weird behavior for other datasets/sizes.